File: packages/rook_sdk/lib/src/data/mapper/availability_status_mappers.dart
import 'package:rook_sdk/src/data/proto/protos.pbenum.dart';
import 'package:rook_sdk/src/domain/enums/availability_status.dart';

extension AvailabilityStatusMappers on AvailabilityStatusProto {
  AvailabilityStatus toDomain() {
    return switch (this) {
      AvailabilityStatusProto.INSTALLED => AvailabilityStatus.installed,
      AvailabilityStatusProto.NOT_INSTALLED => AvailabilityStatus.notInstalled,
      _ => AvailabilityStatus.notSupported,
    };
  }
}

File: packages/rook_sdk/android/src/main/kotlin/io/tryrook/rook_sdk/mapper/AvailabilityStatusMappers.kt
package io.tryrook.rook_sdk.mapper

import com.rookmotion.rook.sdk.domain.enums.AvailabilityStatus
import io.tryrook.rook_sdk.proto.AvailabilityStatusProto

fun AvailabilityStatus.toProto(): AvailabilityStatusProto {
    return when (this) {
        AvailabilityStatus.INSTALLED -> AvailabilityStatusProto.INSTALLED
        AvailabilityStatus.NOT_INSTALLED -> AvailabilityStatusProto.NOT_INSTALLED
        AvailabilityStatus.NOT_SUPPORTED -> AvailabilityStatusProto.NOT_SUPPORTED
    }
}
